src/vs/workbench/contrib/chat/common/codecs/chatbotPromptCodec/chatbotPromptDecoder.ts

```diff
@@ -0,0 +1,45 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { FileReference } from './tokens/fileReference.js';
+import { VSBuffer } from '../../../../../../base/common/buffer.js';
+import { ReadableStream } from '../../../../../../base/common/stream.js';
+import { BaseDecoder } from '../../../../../../base/common/codecs/baseDecoder.js';
+import { Word } from '../../../../../../editor/common/codecs/simpleCodec/tokens/index.js';
+import { SimpleDecoder, TSimpleToken } from '../../../../../../editor/common/codecs/simpleCodec/simpleDecoder.js';
+
+/**
+ * Tokens handled by the `ChatbotPromptDecoder` decoder.
+ */
+export type TChatbotPromptToken = FileReference;
+
+/**
+ * Decoder for the common chatbot prompt message syntax.
+ * For instance, the file references `#file:./path/file.md` are handled by this decoder.
+ */
+export class ChatbotPromptDecoder extends BaseDecoder<TChatbotPromptToken, TSimpleToken> {
```

Comment:
This is a nit, but we don't use `chatbot` anywhere else in this codebase. For consistency with all our other chat features, I would prefer that we renamed `chatbot` to just `chat` in the contents of this file as well as in the directory and file names.

Reply:
renamed